std/str.zig

```diff
@@ -10,6 +11,15 @@ pub fn slice_eql(T: type)(a: []const T, b: []const T) -> bool {
     return true;
 }
 
+pub fn join(T: type)(a: []T, b: []T) -> []T {
+    var result : [a.len + b.len]T = undefined;
+
+    @memcpy(result.ptr, a.ptr, a.len * @sizeof(T));
+    @memcpy((&T)((usize)(result.ptr) + (usize)(a.len * @sizeof(T))), b.ptr, b.len * @sizeof(T));
```

Comment:
You can avoid pointer arithmetic by doing this instead:

```zig
@memcpy(&result[a.len], b.ptr, b.len * @sizeof(T));
```

except that in debug mode, it will crash if `b.len == 0`, so you could do this instead:

```zig
if (@compile_var("is_release") || b.len > 0) {
  @memcpy(&result[a.len], b.ptr, b.len * @sizeof(T));
}
```

The check for release mode ensures that there will be no conditional branch there in release mode, but debug mode needs the conditional branch to avoid a bounds check error on `result[a.len]`.